### ISSUE TITLE
Missing field not detected

### ISSUE THREAD

**Issue by leucospis** [phatcher/CsvReader#12]
I'm using version 3.9.0.0.

In my little test program, the method **CsvReader.CopyCurrentRecordTo(string[] array)** throws LumenWorks.Framework.IO.Csv.MissingFieldCsvException in the case of example CSV 1 (below), but not in the case of example CSV 2 (below). In the case of CSV 2, as far as I can tell, the reader acts as though the third line in the file were this:

"3",""

Should I not be getting a missing field exception in CSV 2?

I'm using these settings:

CsvReader reader = new CsvReader(new StreamReader(fileName), true, ',', '"', '"', '~', ValueTrimmingOptions.None)

DefaultParseErrorAction = ParseErrorAction.ThrowException;
MissingFieldAction = MissingFieldAction.ParseError;
SkipEmptyLines = false;
SupportsMultiline = false;

**Example CSV 1:**

A,B
1,2
3
5,6

**Example CSV 2:**

"A","B"
"1","2"
"3"
"5","6"

**Comment by joshgo** [comment]
@phatcher There is no tag for v3.9.0.0 in the code? Can you tag the corresponding commit?

**Comment by joshgo** [comment]
Neat find. 

If the sample is modified by removing the quotes, then missing field exception gets thrown as expected. If it's quoted, then no exception is thrown. Sounds like a bug.

**Sample A:** (throws exception)

"A","B"
"1","2"
3
"5","6"

**Sample B:** (no exception)

"A","B"
"1","2"
"3"
"5","6"

I created a unit test for this and submitted a PR.

**Comment by joshgo** [comment]
Hi @phatcher,

I added a fix for this issue. I followed the logic of how the unquoted fields are processed, and what seemed to be missing for 'quoted fields' was a check for "end of line". Once a EOL or EOF (end of file) is detected, then we can verify if we have all the fields.

I updated my previous PR, #13. There should be 2 check-ins. The first one is for the unit test, and the other is the fix.

Let me know if you have questions.

Thanks,
Josh
